### PR TITLE
Add an option to set CQL speculative retry policy

### LINKED ISSUE #2563: Allow users to set a speculative retry value on CQL tables
**Describe the feature:**
I would like to be able to set a Speculative Retry value for CQL tables.
https://docs.datastax.com/en/cql-oss/3.1/cql/cql_reference/tabProp.html#tabProp__morespeculativeRetry

**Describe a specific use case for the feature:**
Setting Speculative Retry can help decrease read requests latency, at the expense of extra requests.

### PR DESCRIPTION
**Pull request by cdegroc**
Fixes #2563.

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [X] Is there an issue associated with this PR? Is it referenced in the commit message?
- [X] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [X] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [X] Is your initial contribution a single, squashed commit?

### For code changes:
- [X] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](https://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### CODE REVIEW

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java [hunk 1]
```diff
@@ -367,17 +368,18 @@ String getKeyspaceName() {
 
     @VisibleForTesting
     Map<String, String> getCompressionOptions(final String name) throws BackendException {
-        KeyspaceMetadata keyspaceMetadata = this.session.getMetadata().getKeyspace(this.keyspace)
-            .orElseThrow(() -> new PermanentBackendException(String.format("Unknown keyspace '%s'", this.keyspace)));
-
-        TableMetadata tableMetadata = keyspaceMetadata.getTable(name)
-            .orElseThrow(() -> new PermanentBackendException(String.format("Unknown table '%s'", name)));
```
Comment (cdegroc) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java:374]: Duplicate code. Identical to `getTableMetadata` (same file L381-287)

**Comment by cdegroc** [comment]
Update tests (Codacy, ScyllaDB) and doc. Rebased to make it a single commit.

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java [hunk 2]
```diff
@@ -367,17 +368,18 @@ String getKeyspaceName() {
...
         Object compressionOptions = tableMetadata.getOptions().get(CqlIdentifier.fromCql("compression"));
-
         return (Map<String, String>) compressionOptions;
     }
 
+    @VisibleForTesting
+    Optional<String> getSpeculativeRetry(final String name) throws BackendException {
+        TableMetadata tableMetadata = getTableMetadata(name);
+        Object res = tableMetadata.getOptions().get(CqlIdentifier.fromCql("speculative_retry"));
+        return res == null ? Optional.empty() : Optional.of((String) res);
```
Comment (li-boxuan) [inline janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java:380]: nitpick:
```suggestion
        return Optional.ofNullable((String) res);
```

Comment (li-boxuan) [reply janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java:380]: Actually, it seems "99PERCENTILE" is the default value and this "res" is always non-null, so the "Optional" is probably not needed.

**Review by li-boxuan — APPROVED** [review]
LGTM, I just have a small nitpick

**Review by porunov — APPROVED** [review]
LGTM. Thank you @cdegroc !

File: janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java (hunk 2 above)
Comment (cdegroc) [reply janusgraph-cql/src/main/java/org/janusgraph/diskstorage/cql/CQLStoreManager.java]: Thanks Boxuan. I've updated the code.

**Review by li-boxuan — APPROVED** [review]
LGTM 👍

**Pull request merged**